### PR TITLE
Display dynos for apps in shielded spaces as Shield dynos

### PR DESCRIPTION
**Pull request by jsullivan**
Hi @dickeyxxx and @ransombriggs! For compliance reasons, we need to display the dynos for apps that are in Private Spaces as "Shield-*" sizes. So dynos that would normally display as "Private-M" or "Private-L" will now display as "Shield-M" and "Shield-L" respectively for apps that are in Shielded Private Spaces.

### CODE REVIEW

**Comment by jsullivan** [comment]
cc: @heroku/dogwood-compliance

File: commands/ps/type.js [hunk 1]
```diff
@@ -38,7 +38,19 @@ Types: ${cli.color.yellow(formation.map((f) => f.type).join(', '))}`)
...
   let displayFormation = co.wrap(function * () {
     let formation = yield heroku.get(`/apps/${app}/formation`)
+    let shielded = yield heroku.get(`/apps/${app}`)
+
+    shielded = shielded.space && shielded.space.shield
     formation = sortBy(formation, 'type')
+
+    formation.forEach((d) => {
+      if (shielded && d.size === 'Private-M') {
+        d.size = 'Shield-M'
```
Comment (ryanbrainard) [inline commands/ps/type.js]: can we generalize this as something like `replace(s.size, 'Private-', 'Shield-')?

File: commands/ps/type.js [hunk 2]
```diff
@@ -38,7 +38,19 @@ Types: ${cli.color.yellow(formation.map((f) => f.type).join(', '))}`)
 
   let displayFormation = co.wrap(function * () {
     let formation = yield heroku.get(`/apps/${app}/formation`)
+    let shielded = yield heroku.get(`/apps/${app}`)
```
Comment (ryanbrainard) [inline commands/ps/type.js]: nitpick: can this be `const app` and then assign to `const shielded` below? I just find it weird that `shielded` gets reassigned (first an app and then a boolean)..

File: test/commands/ps/type.js [hunk 3]
```diff
@@ -6,6 +6,11 @@ const cmd = commands.find((c) => c.topic === 'ps' && c.command === 'type')
 const nock = require('nock')
 const expect = require('chai').expect
 
+let app = {
```
Comment (ryanbrainard) [inline test/commands/ps/type.js]: Is is a little weird that this is being used for both the shield and non-shield cases below? maybe it should be a function so callers can do just plain `app()` or `app({shield: true})`?

**Comment by jsullivan** [comment]
@ryanbrainard thanks for the review! Does https://github.com/heroku/heroku-apps/pull/145/commits/374b0305b701e1855e9eb84300cf711472c75231 address all the items from your feedback?

**Comment by ryanbrainard** [comment]
👍

**Comment by jsullivan** [comment]
Any thoughts on this @dickeyxxx or @ransombriggs?

**Comment by ransombriggs** [comment]
@jsullivan @dickeyxxx what are the compliance requirements?  Depending on the exact requirements, I would rather this be approached differently in one of two ways.

If it is important that the name be `Shield` then I would suggest that the change be made in core services API instead so that there is not confusion between how the API / dashboard / CLI behave and that they all be consistent.

If the requirement is just that we include if the dynos are shielded or not, then I would add another column to the table to indicate that the dyno is shielded or not.

**Comment by jsullivan** [comment]
@ransombriggs this is just a cosmetic change for now as the name "Shield" for representing these types of Private Spaces has not been settled on. It's a pain to integrate all of this with Vault, API, etc. for dyno types that aren't a for-sure thing yet. For now we just need a visual differentiation between what we call dynos in regular private spaces and what we call them in these shielded spaces.

This change is [already shipped on the dashboard](https://github.com/heroku/dashboard/pull/5227).

**Comment by ransombriggs** [comment]
@jsullivan This PR would not work as-is because there is not symmetry between dyno display and dyno scaling. For instance, commands like this will not work, because API does not know about `Shield`.
```
heroku ps:scale web=2:Shield-2X
```
Teaching API about `Shield-2X` would be the most obvious solution, but it sounds like it is not possible at this time

I think it would be best to add a `Shield` boolean column to API's dyno table at this time, and use `Private` names.  We could use the `Shield` boolean to add descriptive text and icons, as appropriate, to the dashboard and CLI for the associated plans.

**Comment by jsullivan** [comment]
hey @ransombriggs that's good feedback. To keep the CLI consistent with the dashboard experience, I'd like to just update the `scale` command to use the same translation of `Shield-*` to `Private-*` when the command is issued on apps in shielded spaces. That'd make it easiest to change the CLI and dashboard when we finally settle on a long-term solution for these different private space types.

As I said earlier, the current dashboard experience that product has approved is for shielded private spaces to show a "Shield-M" and "Shield-L" dyno in shielded spaces, not a Private dyno with some descriptive text. When a user in dashboard wants to scale one of their dynos in a private space, they choose which "Shield" dyno and then scale it. For consistency I think it best to keep that pattern here in the CLI for the time being. What are your thoughts on this?

**Comment by ransombriggs** [comment]
@jsullivan could you make the changes I discussed to `heroku ps:scale` and I can review / merge?  If it is complicated, let me know and I can do it, just do not have the time to look at it right now.

**Comment by jsullivan** [comment]
@ransombriggs for sure, I just wanted to run it by you first before I made the changes. Sounds like we have a path forward for now. Thanks so much!

**Comment by ransombriggs** [comment]
@jsullivan please add a test for [these changes](https://github.com/heroku/heroku-apps/pull/145/commits/93d6aac51a4ae39338757ea7ce054cea4ad0d28c)

**Comment by ryanbrainard** [comment]
👍

**Comment by jsullivan** [comment]
@ransombriggs tests are added and passing. Can this go out now?

**Pull request merged**